internal/api/auth/basic_auth.go

```diff
@@ -12,6 +13,44 @@ const (
 	realm                = `Basic realm="restricted"`
 )
 
+type authMode int
+
+const (
+	NoAuth authMode = iota
+	LDAPMode
+)
+
+type BasicAuth struct {
+	ldapAuth *LDAPAuth
+	mode     authMode
+}
+
+func NewBasicAuth() BasicAuth { return BasicAuth{mode: NoAuth} }
+
+func (b *BasicAuth) SetMode(mode authMode) { b.mode = mode }
```

Comment:
I am not sure if constructing the object in multiple steps really make sense: do we expect the user to change the mode after the object creation ?

Reply:
Done